semgrep-core/src/analyzing/CFG_build.ml

```diff
@@ -261,14 +261,28 @@ and cfg_stmt_list state previ xs =
            | CfgLabel label -> (previ, label :: labels))
          (previ, [])
   in
-  match (lasti_opt, labels) with
-  | Some lasti, l :: ls ->
-      (* If we had labels at the end of our stmt list, we create a dummy node to assign them to,
-         and connect it to the last node we looked at
-      *)
-      let dummyi = state.g#add_node { n = NOther Noop } in
+  match labels with
+  | l :: ls ->
+      (* If we have labels at the end of our list of stmt, we create a dummy
+       * node to assign them to.
+       *
+       *     lasti -> label: dummy
```

Comment:
I've found this 'lasti -> label: dummy' confusing, because actually what this new code does is that it
creates the dummy node even if there is no lasti right?

Reply:
Yeah that's right, I edited the comment to remove that. It was probably not adding much anyways.